.github/workflows/_build_container.yml

```diff
@@ -60,19 +70,7 @@ jobs:
             MAX_JOBS=32
             ALIGNER_COMMIT=${{ github.event.pull_request.head.sha || github.sha }}
           cache-from: |
-            nemoci.azurecr.io/nemo_aligner_container:${{ github.event.pull_request.number || 'buildcache' }}
-            nemoci.azurecr.io/nemo_aligner_container:buildcache
-          cache-to: type=inline
-          tags: nemoci.azurecr.io/nemo_aligner_container:${{ github.run_id }}
-
-      - name: Update PR image
-        if: github.event_name == 'pull_request'
-        run: |
-          docker tag nemoci.azurecr.io/nemo_aligner_container:${{ github.run_id }} nemoci.azurecr.io/nemo_aligner_container:${{ github.event.pull_request.number }}
-          docker push nemoci.azurecr.io/nemo_aligner_container:${{ github.event.pull_request.number }}
-
-      - name: Update buildcache image
-        if: github.ref == 'refs/heads/main'
-        run: |
-          docker tag nemoci.azurecr.io/nemo_aligner_container:${{ github.run_id }} nemoci.azurecr.io/nemo_aligner_container:buildcache
-          docker push nemoci.azurecr.io/nemo_aligner_container:buildcache
\ No newline at end of file
+            nemoci.azurecr.io/nemo_aligner_container_cache:${{ github.event.pull_request.number || 'noop' }}
+            nemoci.azurecr.io/nemo_aligner_container_cache:main
+          cache-to: ${{ env.CACHE }}
+          tags: nemoci.azurecr.io/nemo_aligner_container:${{ github.run_id }}
```

Comment:
Does docker build always pull the cache image even if the dockerfile is identical? If so, is it possible to speed up the docker build by doing something like this (pseudocode):
```sh
if [[ ! image_tag_exists "nemoci.azurecr.io/nemo_aligner_container:PR-${pr}-$(md5sum Dockerfile) ]] || [[ $CACHEBUST ]]; then
  # do build
else
  # noop
fi
```

Reply:
Interesting thought. If we were to use `sed` instead of `build-arg`, this might work. But we'd also be working around Docker's native caching mechanism, and bugs around caching can unfold in an extremely unpleasant way - I'm not sure if we want to go that route yet?

Reply:
And then both drivers have the shared property of requiring all layers to be present on the host system for building an updated image (since overlay-fs layers are chained by the merge-dir to their parent)